libs/scheduler/aws.go

```diff
@@ -62,7 +66,13 @@ func (job *Job) PopulateAwsCredentialsEnvVarsForJob() error {
 			log.Printf("Failed to get keys from role: %v", err)
 			return fmt.Errorf("Failed to get (state) keys from role: %v", err)
 		}
-
+		} else {
+			job.StateEnvVars, err = populateKeys(job.StateEnvVars, *job.StateEnvProvider)
+			if err != nil {
+				log.Printf("Failed to get keys from role (StateEnvProvider): %v", err)
+				return fmt.Errorf("Failed to get (state) keys from role: %v", err)
+			}
+		}
```

Comment:
So this is not a challenge and I'm probably wrong because I haven't looked at this code in a while but from memory I remmber that the reason for using "backend configuration" passed as arguments rather than env variables was to tell terraform to pull S3 state from an account different from the account where it is creating resources. 

From my basic tests it is not possible to achieve this with env variables during the init phase. SO, I wonder if this entire else block is needed at all. Because if terragrunt has its own way to pass the same config then maybe we ignore the else entirely.

I'm pretty sure I missed something but very useful to discuss this on here for historical context of the code